docs/rfc/rfc-6.adoc

```diff
@@ -0,0 +1,219 @@
+:toc: macro
+
+= RFC 6: Transaction incentives
+
+:icons: font
+:numbered:
+toc::[]
+
+== Goal
+The goal of this proposal is to reduce on-chain cost and complexity related to
+transaction submitter responsibilities, reduce the cost of operating a client,
+and provide tools to punish free-riders not operating their nodes properly.
+
+== Overview
+The scheme assumes that the majority of the group is honest and is interested in
+helping DAO to eliminate free-riders from the network while getting nothing in
+return. The group will vote to mark someone as ineligible for rewards if they
+consistently do not participate in off-chain heartbeats or if they skip their
+responsibilities for submitting on-chain transactions. 
+
+A real-world analogy would be team members discussing one of them is not
+contributing to the work at all so the entire team must do the work of that
+free-rider and have the team report this fact to the manager.
+
+All the responsibilities that do not have to be done by a group will be
+delegated to opt-in maintainers and third parties to reduce the cost of
+operating a node.
+
+== In depth
+
+=== Transaction types
+
+On-chain transactions can be divided into three types:
+
+- Transactions that can only be submitted by group members, given that only the
+  group members have the knowledge about input parameters. For example, submit
+  DKG result transaction for ECDSA wallet or submit relay entry transaction for
+  the random beacon.
+  
+- Transactions that can be submitted by anyone, usually submitted often and
+  mostly belonging to a happy path. Never leading to slashing For example,
+  submit sweep proof or submit redemption proof transaction for tBTC v2. 
+
+- Transactions that can be submitted by anyone, submitted rarely or never, and
+  usually leading to slashing. For example, challenge DKG result for ECDSA
+  wallet or report fraud proof transaction for tBTC v2.
+
+For the first group of transactions, we will not enforce the order on-chain and
+we will expect off-chain clients to respect the submission order. If some client
```

Comment:
I would hyperlink "submission order" to its definition here to avoid confusion and give it its own section.

Reply:
Not sure if I follow, should we make this paragraph a definition of a submission order and link from _Submit + approve DKG result_ to it?

Reply:
Here, were saying that we expect the off-chain clients to respect the submission order, but at this point in the document, the submission order isn't defined yet. My suggestion was to create a new header like

```
[[submission-order]]
===== Submission Order
Whenever it is prudent to not have members of the signing group race to submit a transaction and waste gas, they agree off-chain to submit in an order, defined as follows...
```

and then link to that section whenever you talk about the submission order anywhere else